new/analyse/spec/add-dependencies.spec.ts

```diff
@@ -93,4 +88,122 @@ describe('addDependencies Module', () => {
       expect(result).toEqual(expectedEnv);
     });
   });
+
+  describe('with rlsr.json data', () => {
+    let result: Env;
+    let dependencies: RelatedPackageDependsOn[];
+
+    beforeAll(() => {
+      const env: Env = {
+        ...envWithConfig,
+        status: {
+          packages: {
+            mockPackageUnderTest: {
+              version: '1.0.0',
+              dependencies: {
+                mockPreviouslyAvailable: {
+                  name: 'mockPreviouslyAvailableButOverwritten',
+                  type: 'default',
+                  range: '1 - 2',
+                },
+                mockPreviouslyAvailableButOverwritten: {
+                  name: 'mockPreviouslyAvailableButOverwritten',
+                  type: 'default',
+                  range: '1 - 3',
+                },
+                mockPreviouslyAvailableButNew: {
+                  name: 'mockPreviouslyAvailableButOverwritten',
+                  type: 'default',
+                  range: '1 - 4',
+                },
+              },
+            },
+            mockPreviouslyAvailable: {
+              version: '1.0.0',
+              dependencies: {},
+            },
+            mockPreviouslyAvailableButOverwritten: {
+              version: '1.0.0',
+              dependencies: {},
+            },
+            mockPreviouslyAvailableButNew: {
+              version: '1.0.0',
+              dependencies: {},
+            },
+          },
+        },
+        packages: {
+          mockPackageUnderTest: createMockPackage('mockPreviouslyAvailable', {
+            mockPreviouslyAvailable: '*',
+            mockPreviouslyAvailableButOverwritten: '2 - 3',
+            mockPreviouslyAvailableButNew: 'new',
+            mockPreviouslyNotAvailable: '*',
+          }),
+          mockPreviouslyAvailable: createMockPackage('mockPreviouslyAvailable'),
+          mockPreviouslyAvailableButOverwritten: createMockPackage(
+            'mockPreviouslyAvailableButOverwritten'
+          ),
+          mockPreviouslyAvailableButNew: createMockPackage(
+            'mockPreviouslyAvailableButNew'
+          ),
+          mockPreviouslyNotAvailable: createMockPackage(
+            'mockPreviouslyNotAvailable'
+          ),
+        },
+      };
+      result = addDependencies(env) as Env;
+      dependencies = result?.packages?.mockPackageUnderTest?.dependsOn ?? [];
+    });
+    it('has a result', () => {
+      expect(result).toBeDefined();
+    });
+    it('takes the dependency from rlsr.json if it is `*`', () => {
+      const found = dependencies.find(
+        (dependency) => dependency.name === 'mockPreviouslyAvailable'
+      );
+      expect(found).toBeDefined();
+      expect(found?.range).toEqual('1 - 2');
```

Comment:
You could use `expect.objectContaining(object)` as well (https://jestjs.io/docs/expect#expectobjectcontainingobject) to test in one assertion multiple properties 

```javascript
expect(result?.packages?.mockPackageUnderTest?.dependsOn).toEqual(expect.objectContaining({
  name: 'mockPreviouslyAvailable'
  range: '1 - 2'
}))
```

Reply:
But it's an array. Doesn't seem to work according to their specs. Also `arrayContaining` doesn't work.